ApnsPHP/Message/LiveActivity.php

```diff
@@ -0,0 +1,259 @@
+     */
+    private array|object $state;
+
+    /**
+     * Attributes at the start of an activity
+     * @var array|object
+     */
+    private array|object $attributes;
+
+    /**
+     * Type of attributes
+     * @var string
+     */
+    private string $attributes_type;
+
+    /**
+     * Event for the activity
+     * @var LiveActivityEvent
+     */
+    private LiveActivityEvent $event;
+
+    /**
+     * Time when the activity will become stale
+     * @var int
+     */
+    private int $stale_time;
+
+    /**
+     * Time when the activity will dismiss itself
+     * @var int
+     */
+    private int $dismiss_time;
+
+    /**
+     * Constructor.
+     *
+     * @param string|null $deviceToken Recipients device token (optional).
+     */
+    public function __construct(?string $deviceToken = null)
+    {
+        parent::__construct($deviceToken);
+        parent::setPushType(PushType::LiveActivity);
+    }
+
+    /**
+     * Get the payload dictionary.
+     *
+     * @return array<string,mixed> The payload dictionary.
+     */
+    public function getPayloadDictionary(): array
+    {
+        $payload = parent::getPayloadDictionary();
+
+        $payload[self::APPLE_RESERVED_NAMESPACE]['event'] = $this->event->value;
+        $payload[self::APPLE_RESERVED_NAMESPACE]['timestamp'] = time();
+        if (isset($this->state)) {
+            $payload[self::APPLE_RESERVED_NAMESPACE]['content-state'] = $this->state;
+        }
+        if (isset($this->stale_time)) {
+            $payload[self::APPLE_RESERVED_NAMESPACE]['stale-date'] = $this->stale_time;
+        }
+        if (isset($this->dismiss_time)) {
+            $payload[self::APPLE_RESERVED_NAMESPACE]['dismissal-date'] = $this->dismiss_time;
+        }
+
+        if ($this->event !== LiveActivityEvent::Start) {
+            return $payload;
+        }
+
+        if (isset($this->attributes) && isset($this->attributes_type)) {
+            $payload[self::APPLE_RESERVED_NAMESPACE]['attributes-type'] = $this->attributes_type;
+            $payload[self::APPLE_RESERVED_NAMESPACE]['attributes'] = $this->attributes;
+        }
+
+        return $payload;
+    }
+
+    /**
+     * Set a topic
+     *
+     * @throws UnexpectedValueException if the topic is invalid for a live activity.
+     *
+     * @param string $topic
+     *
+     * @return void
+     */
+    public function setTopic(string $topic): void
+    {
+        if (!str_contains($topic, '.push-type.liveactivity')) {
+            throw new UnexpectedValueException("Topic '$topic' does not include '.push-type.liveactivity'!");
+        }
+
+        parent::setTopic($topic);
+    }
+
+    /**
+     * Set a push type
+     *
+     * @throws AssertionError Since the push type is tied to the class
+     *
+     * @param PushType $pushType
+     *
+     * @return void
+     */
+    public function setPushType(PushType $pushType): void
+    {
+        throw new AssertionError('Push type is enforced by the class!');
```

Comment:
This seems odd. ```AssertionError is thrown when an assertion made via [assert()](https://www.php.net/manual/en/function.assert.php) fails.``` doesn't really fit.

I'd go with `DomainException` or `RuntimeException` here.